common/src/test/java/org/apache/uniffle/common/netty/EncoderAndDecoderTest.java

```diff
@@ -0,0 +1,174 @@
+import io.netty.buffer.Unpooled;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import org.apache.uniffle.common.ShuffleBlockInfo;
+import org.apache.uniffle.common.ShuffleServerInfo;
+import org.apache.uniffle.common.exception.RssException;
+import org.apache.uniffle.common.netty.protocol.NettyProtocolTestUtils;
+import org.apache.uniffle.common.netty.protocol.RpcResponse;
+import org.apache.uniffle.common.netty.protocol.SendShuffleDataRequest;
+import org.apache.uniffle.common.rpc.StatusCode;
+import org.apache.uniffle.common.util.NettyUtils;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class EncoderAndDecoderTest {
+  private EventLoopGroup bossGroup;
+  private EventLoopGroup workerGroup;
+  private ChannelFuture channelFuture;
+  private static final SendShuffleDataRequest DATA_REQUEST = generateShuffleDataRequest();
+  private static final String EXPECTED_MESSAGE = "test_message";
+  private static final long REQUEST_ID = 1;
+  private static final StatusCode STATUS_CODE = StatusCode.SUCCESS;
+  private static final int PORT = 12345;
+
+  static class MockResponseHandler extends ChannelInboundHandlerAdapter {
+    @Override
+    public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+      if (msg instanceof RpcResponse) {
+        RpcResponse rpcResponse = (RpcResponse) msg;
+        assertEquals(REQUEST_ID, rpcResponse.getRequestId());
+        assertEquals(STATUS_CODE, rpcResponse.getStatusCode());
+        assertEquals(EXPECTED_MESSAGE, rpcResponse.getRetMessage());
+      } else if (msg instanceof SendShuffleDataRequest) {
+        SendShuffleDataRequest sendShuffleDataRequest = (SendShuffleDataRequest) msg;
+        assertTrue(NettyProtocolTestUtils.compareSendShuffleDataRequest(sendShuffleDataRequest, DATA_REQUEST));
+        sendShuffleDataRequest.getPartitionToBlocks().values().stream().flatMap(Collection::stream)
+            .forEach(shuffleBlockInfo -> shuffleBlockInfo.getData().release());
+        sendShuffleDataRequest.getPartitionToBlocks().values().stream().flatMap(Collection::stream)
+            .forEach(shuffleBlockInfo -> assertEquals(0,shuffleBlockInfo.getData().refCnt()));
+        RpcResponse rpcResponse = new RpcResponse(REQUEST_ID, STATUS_CODE, EXPECTED_MESSAGE);
+        ctx.writeAndFlush(rpcResponse);
+      } else {
+        throw new RssException("receive unexpected message!");
```

Comment:
I think we can add a class of netty transport exceptions in the followup pr, which may better distinguish these exceptions.